mrp_merge_productions/mrp_production.py

```diff
@@ -0,0 +1,320 @@
+                    to_write = updates.setdefault(
+                        mergeable.id,
+                        {"product_qty": mergeable.product_qty,
+                         "product_uos_qty": mergeable.product_uos_qty}
+                    )
+
+                    _logger.debug("Merging move %s into move %s (%d %s)",
+                                  to_merge.id, mergeable.id,
+                                  to_merge.product_qty,
+                                  to_merge.product_id.name)
+                    to_write["product_qty"] += to_merge.product_qty
+                    if to_merge.product_uos_qty:
+                        to_write["product_uos_qty"] += to_merge.product_uos_qty
+                    to_cancel.append(to_merge.id)
+
+                    # Get the next in the chain
+                    to_merge = to_merge.move_dest_id
+                    mergeable = mergeable.move_dest_id
+                    if to_merge and mergeable:
+                        _logger.debug(
+                            "Next moves [%d]:%d %s, [%d]:%d %s",
+                            to_merge.id, to_merge.product_qty,
+                            to_merge.product_id.name,
+                            mergeable.id, mergeable.product_qty,
+                            mergeable.product_id.name,
+                        )
+                        if not moves_mergeable(to_merge, mergeable):
+                            break
+                        # Find a closing procurement order on the moves
+                        proc_order = proc_order_obj.search(cr, uid, [
+                            ('move_id', 'in', (to_merge.id, mergeable.id)),
+                            ('close_move', '=', True),
+                        ])
+                        if proc_order:
+                            break
+                        # Don't allow a change of product or unit
+                        if to_merge.product_id.id != orig_product:
+                            break
+                        if id_of(to_merge.product_uom) != orig_uom:
+                            break
+
+                # Found a target, stop searching
+                break
+        else:
+            raise orm.except_orm(
+                _("Unable to merge 'To Produce' move"),
+                _("No candidate to merge move %s") % (to_merge.id, ),
+            )
+
+
+class mrp_production(orm.Model):
+    _inherit = 'mrp.production'
+
+    def _check_productions_mergeable(self, productions):
+        if any(production.state in ('done', 'cancel', 'in_production')
+               for production in productions):
+            raise orm.except_orm(
+                _("Invalid Selection"),
+                _("You cannot merge manufacturing orders that are done, "
+                  "in production or cancelled."),
+            )
+
+        unique_attrs = (
+            ("product_id", _("Product")),
+            ("bom_id", _("Bill of Materials")),
```

Comment:
Should'nt you consider the routing too ?
I'd not watn to merge MOs that are not made using the same manufacuturing process